contracts/optimistic-ethereum/OVM/bridge/OVM_BaseCrossDomainMessenger.sol

```diff
@@ -46,11 +48,10 @@ contract OVM_BaseCrossDomainMessenger is iOVM_BaseCrossDomainMessenger {
             messageNonce
         );
 
-        _sendXDomainMessage(xDomainCalldata, _gasLimit);
-
         messageNonce += 1;
         sentMessages[keccak256(xDomainCalldata)] = true;
 
+        _sendXDomainMessage(xDomainCalldata, _gasLimit);
```

Comment:
I think that this order does not matter, especially because the `_sendXDomainMessage` function only talks to trusted code, but checks-effects-interactions pattern still good to uphold 👍